app/models/user.rb

```diff
@@ -520,6 +521,12 @@ def ensure_default_role
     end
   end
 
+  def ensure_admin_password_changed_by_admin
+    if (self.admin && !User.current.admin) && password_hash_changed?
+      errors.add :password, _("You can't change administrator password")
```

Comment:
`cannot be changed by a non-admin user`? (Partial sentences work better with AR errors I think, and avoids referring in the second person.)